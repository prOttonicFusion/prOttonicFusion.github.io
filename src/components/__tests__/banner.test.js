import React from 'react'
import { render } from '@testing-library/react'
import { withQuerySnapshot } from 'gatsby-plugin-testing'
import Banner from '../banner'

// https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
})

describe('Page banner', () => {
    it('renders root page banner correctly', withQuerySnapshot(() => {
        const { container } = render(
            <Banner location="/"/>,
        )
        expect(container.firstChild).toMatchSnapshot()
    }))

    it('renders child page banner correctly', withQuerySnapshot(() => {
        const { container } = render(
            <Banner location="/some/page"/>,
        )
        expect(container.firstChild).toMatchSnapshot()
    }))
})
