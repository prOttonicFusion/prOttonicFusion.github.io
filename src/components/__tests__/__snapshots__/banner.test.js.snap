// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page banner renders child page banner correctly 1`] = `
<div
  class="page-banner"
>
  <nav
    class="site-nav navbar navbar-expand-md navbar-dark bg-none"
  >
    <div
      class="container-fluid"
    >
      <a
        class="site-title navbar-brand"
        href="/"
      >
        Otto Lindblom
      </a>
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="nav-toggle navbar-toggler collapsed"
        type="button"
      >
        <svg
          class="nav-toggle-icon"
          fill="#424242"
          height="32px"
          viewBox="0 0 20 20"
          width="32px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-top"
        id="offcanvasNavbar-expand"
      >
        <div
          class="offcanvas-header"
        >
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="offcanvas-body"
        >
          <div
            class="justify-content-end flex-grow-1 pe-3 navbar-nav"
          >
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#about"
              href="/#about"
            >
              About
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#projects"
              href="/#projects"
            >
              Projects
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/brief-cv"
              href="/brief-cv"
            >
              Brief CV
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#contact"
              href="/#contact"
            >
              Contact
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Page banner renders root page banner correctly 1`] = `
<div
  class="page-banner"
>
  <nav
    class="site-nav navbar navbar-expand-md navbar-dark bg-none"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="nav-toggle navbar-toggler collapsed"
        type="button"
      >
        <svg
          class="nav-toggle-icon"
          fill="#424242"
          height="32px"
          viewBox="0 0 20 20"
          width="32px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-top"
        id="offcanvasNavbar-expand"
      >
        <div
          class="offcanvas-header"
        >
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="offcanvas-body"
        >
          <div
            class="justify-content-end flex-grow-1 pe-3 navbar-nav"
          >
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#about"
              href="/#about"
            >
              About
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#projects"
              href="/#projects"
            >
              Projects
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/brief-cv"
              href="/brief-cv"
            >
              Brief CV
            </a>
            <a
              class="custom-nav-link nav-link"
              data-rr-ui-event-key="/#contact"
              href="/#contact"
            >
              Contact
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
